pkg/roles/service.go

```diff
@@ -112,18 +114,33 @@ func (s *Service) CreateRole(ctx context.Context, userID, ID string) error {
 	ctx, span := s.tracer.Start(ctx, "roles.Service.CreateRole")
 	defer span.End()
 
-	// TODO @shipperizer will we need also the can_view?
+	// TODO @shipperizer @barco will we need also the can_edit, can_delete?
 	// does creating a role mean that you are the owner, therefore u get all the permissions on it?
 	// right now assumption is only admins will be able to do this
 	// potentially changing the model to say
 	// `define can_view: [user, user:*, role#assignee, group#member] or assignee or admin from privileged`
 	// might sort the problem
 
 	// TODO @shipperizer offload to privileged creator object
+	role := fmt.Sprintf("role:%s", ID)
+	user := fmt.Sprintf("user:%s", userID)
+
 	err := s.ofga.WriteTuples(
 		ctx,
-		*ofga.NewTuple(fmt.Sprintf("user:%s", userID), ASSIGNEE_RELATION, fmt.Sprintf("role:%s", ID)),
-		*ofga.NewTuple(authorization.ADMIN_PRIVILEGE, "privileged", fmt.Sprintf("role:%s", ID)),
+		*ofga.NewTuple(user, ASSIGNEE_RELATION, role),
+		*ofga.NewTuple(authorization.ADMIN_PRIVILEGE, "privileged", role),
+	)
+
+	if err != nil {
+		s.logger.Error(err.Error())
+		return err
+	}
+
+	// after successful role creation, `can_view` to user:* needs to be assigned
+	// explicitly for it to work
+	err = s.ofga.WriteTuples(
+		ctx,
+		*ofga.NewTuple(ALL_USERS, CAN_VIEW_RELATION, role),
 	)
```

Comment:
I would say we should bundle all the writes in one go on the above, chances that fails but this doesn't are quite small unless we have a conflict which we should check somehow beforehand